src/audio/base_fw.c

```diff
@@ -240,14 +240,18 @@ static uint32_t basefw_set_system_time(uint32_t param_id,
 				       uint32_t data_offset,
 				       const char *data)
 {
+	uint64_t current_dsp_cycle;
+	uint64_t current_dsp_time;
+
 	/* TODO: configurate time to logging subsystem */
 	if (!(first_block && last_block))
 		return IPC4_INVALID_REQUEST;
 
 	global_system_time_info.host_time.val_l = ((const struct ipc4_system_time *)data)->val_l;
 	global_system_time_info.host_time.val_u = ((const struct ipc4_system_time *)data)->val_u;
 
-	uint64_t current_dsp_time = sof_cycle_get_64();
+	current_dsp_cycle = sof_cycle_get_64();
+	current_dsp_time = k_cyc_to_us_floor64(current_dsp_cycle);
```

Comment:
Why do you need `current_dsp_cycle`?
```c
current_dsp_time = k_cyc_to_us_floor64(sof_cycle_get_64());
```
is not working?

Reply:
we save current_dsp_cycle to dsp_time in the following code

Reply:
I see, but in this patch it is not needed? At the end it does not going to matter, but I usually prefer to not introduce stale variables that is going to be used by upcoming patch without pointing out that `current_dsp_cycle` is introduced for the next patch.

Reply:
updated